Mathlib/RingTheory/Ideal/IsPrincipal.lean

```diff
@@ -0,0 +1,299 @@
+/-
+Copyright (c) 2018 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Chris Hughes, Mario Carneiro
+-/
+import Mathlib.Algebra.Algebra.Operations
+
+/-!
+# Principal Ideals
+-/
+
+namespace Ideal
+
+variable {R : Type*}
+
+section Semiring
+
+variable [Semiring R]
+
+@[simp]
+theorem zero_eq_bot : (0 : Ideal R) = ⊥ :=
```

Comment:
Yes, it was in `Operations`. I moved it here because I need it here and, after all, zero is a principal ideal. It cannot be in `Basic` because of imports and I don't want to change `Basic` imports.

Reply:
I think this can go under `Submodule.zero_eq_bot`?

Reply:
Okay after reading the remaining of the file, maybe add a new file `RingTheory/Ideal/Lattice`?

Reply:
Yeah, I think I'm going to put this PR on hold. There is first a need to split `Operations` into several smaller files first for this to work.